sklearn/impute/_knn.py

```diff
@@ -115,6 +117,15 @@ class KNNImputer(_BaseImputer):
            [8. , 8. , 7. ]])
     """
 
+    _parameter_constraints = {
+        "missing_values": [Real, Integral, str, None],
```

Comment:
This case is a bit tricky indeed. We should check for values such as `np.nan` or `pd.NA`. The latter is not an easy one since we cannot import pandas to not depend on it.

We will be discussing how to handle this case in the following issue: https://github.com/scikit-learn/scikit-learn/issues/23919

The idea will be to create a specific type as `"verbose"` or `"boolean"`.

Reply:
> This case is a bit tricky indeed. We should check for values such as np.nan or pd.NA. The latter is not an easy one since we cannot import pandas to not depend on it.

@glemaitre I think it does not apply to KNNImputer since the docstring says
```
For pandas' dataframes with nullable integer dtypes with missing values, `missing_values`
should be set to np.nan, since `pd.NA` will be converted to np.nan.
```
But I agree than we need to find a solution for other imputers